### DISCUSSION TITLE
On rendering two images with zoom enabled in full screen with react-full-screen library

### DISCUSSION THREAD

**Discussion by suganya27302** [openseadragon/openseadragon#2495] [Q&A]
on rendering two images in full screen, the rendered images are zoomed to the center part, we need to pan to see the other parts of the image, Is this the exact behavior? or we can render the actual size of the image and zoom?

**Comment by iangilman** [comment]
What are you trying to accomplish? The user can pan and zoom as they please. Are you trying to programmatically pan and zoom? If so, there are features for that in the Viewport:

https://openseadragon.github.io/docs/OpenSeadragon.Viewport.html

Depending on what you're looking for, `fitBounds` is a good general-purpose tool. You'll need to understand viewport coordinates:

https://openseadragon.github.io/examples/viewport-coordinates/

**Reply by suganya27302** [reply]
No, I don't want to programmatically pan and zoom. If the width of the OSD div is half of the height ex: width:565px, height:958px or the height and width have only a smaller difference Example: width:850px, height:952px, in such case, the image is rendered in a `zoomed in` view. The height and width of the OSD div are calculated dynamically based on the screen size.

 In a full screen, I was rendering two images vertically, the images are rendered in zoomed-in view.

![image](https://github.com/openseadragon/openseadragon/assets/77053019/84b2a1fb-ada8-4292-a96c-864c17829a54)

 If I render 4 images in 2X2, the images are not rendered in zoomed-in view, it renders the full image as expected. Here the height is half of the width, Ex:  width:850px, height:477px

![image](https://github.com/openseadragon/openseadragon/assets/77053019/fd6d0b09-2a53-4023-9977-03fa85d40398)

Why is this happening? In all cases, it should behave the same right? But in some cases, it is in zoomed-in view, and in some cases it renders the full image

**Reply by iangilman** [reply]
So this is just where OSD put things without any user interaction? Looks like you're using the `homeFillsViewer` option... Is that true?

Btw, you mentioned fullscreen, but if you have a bunch of viewers at the same time I assume none of them are actually in fullscreen mode. Do you just mean that you're filling the page with them?

**Reply by pearcetm** [reply]
> Btw, you mentioned fullscreen,

Based on the top screenshot which has the box to "Press [Esc] to exit full screen" looks like it is the native browser full-screen mode, not OSD's version of full screen.

**Reply by iangilman** [reply]
@pearcetm Oh, good point! Thank you.
